### DISCUSSION TITLE
Prisma ORM and SqlServer: I can't connect to the database even using "trustServerCertificate=true"

### DISCUSSION THREAD

**Discussion by antonioedinadson** [prisma/prisma#16953] [Q&A]
I have a local server running **Ubuntu Server,** where my API built on **NodeJS and Prisma ORM** is hosted and the **SQL Server database** is hosted on another local server that runs **Windows Server 2019 .**

When I'm developing on my local machine running **Windows 11 PRO,** Prisma ORM connects quickly and can map the entire database without any errors.

But when I try to map the database on the Linux server (npx prisma db pull) I get the error:

> Error: P1011
> Error opening a TLS connection: The TLS settings didn't allow the connection to be established. Please review your connection string. (error: unexpected EOF)

Some people said that it was just adding the "trustServerCertificate=true" in the "DatabaseURL" but I keep getting the same error:

> Error: P1011
> Error opening a TLS connection: The TLS settings didn't allow the connection to be established. Please review your connection string. (error: unexpected EOF)

**I pinged and the LINUX server can see the WINDOWS server**

> mey .env

```
PORT=3001
DATABASE_URL="sqlserver://MY_IP:1433;user=MY_USER;password=MY_PASS;TrustServerCertificate=true;database=MY_DB"
```

> my schema.prisma
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}
```
### Without using PRISMA it works perfectly on Ubuntu Server.

**Comment by nurul3101** [comment]
Hello @AntonioEdinadson 👋 

Did you try the suggestions mentioned in this [Stack Overflow Answer](https://stackoverflow.com/questions/70928150/authentication-failed-for-user-when-running-prisma-db-pull-on-a-sql-server-dat)? The answer discusses the same issue that you are facing.

On a side note, I would recommend you use English for communication, as everyone mostly is proficient in English compared to other languages. 😄

**Reply by antonioedinadson** [reply]
Hello @nurul3101, 

I followed this Stack Overflow procedure, and it didn't work for me.
All right, I did the translation into English.

**Reply by nurul3101** [reply]
Is it happening on latest prisma version?

**Reply by antonioedinadson** [reply]
The error was in version 4.6.1, I updated to version 4.8.0 and it remains the same.

**Reply by nurul3101** [reply]
Can you try adding `Encrypt=True` in your connection string?
It will force the connection to use TLS encryption, which may resolve the issue if it is related to compatibility.

**Reply by antonioedinadson** [reply]
Hi nurul3101,

Even adding `"Encrypt=True"` in the connection string the error continues.

See the video below where the error is shown.
https://drive.google.com/file/d/1UOpgrRIIXV3J6InJcmmUb88mJBL9hUhb/view?usp=share_link

**Reply by nurul3101** [reply]
Can you please create a [Bug Report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) so that our engineering team can have a look and provide insights on what could be the issue?

**Comment by imsam** [comment]
I'm seeing very similar behaviour developing a node.js-based app, connecting to SQL Server. Everything runs fine locally on my macOS machine, but as soon as I try to run it in a Docker container (running node/alpine - I've tried the node images for node 16, 18 and 20) I'm getting unexpected EOF errors for my connection string.

`Error opening a TLS connection: The TLS settings didn't allow the connection to be established. Please review your connection string. (error: unexpected EOF)`

If I try to run prisma db pull during the build step of my deployment, I see the following:

```
Prisma schema loaded from prisma/schema.prisma
Environment variables loaded from .env
Datasource "db": SQL Server database

- Introspecting based on datasource defined in prisma/schema.prisma
✖ Introspecting based on datasource defined in prisma/schema.prisma

Error: P1011

Error opening a TLS connection: The TLS settings didn't allow the connection to be established. Please review your connection string. (error: unexpected EOF)
```

This particular deployment was working just fine a month or two ago, so it feels like an update to something recently could be the cause. I'm still investigating whether any infrastructure changes could be responsible, but was has there been any progress on this since January? Was a Bug Report submitted, or would you like me to submit one?

My (redacted) connection string follows the below format:
`DATABASE_URL="sqlserver://SERVERIP\INSTANCENAME;database=DBNAME;user=USERNAME;password=PASSWORD;trustServerCertificate=true;"`

**Reply by nurul3101** [reply]
@imsam Can you please open a new [GitHub Discussion](https://github.com/prisma/prisma/discussions/new) so that we can investigate?

Thanks!

**Reply by imsam** [reply]
My issue, which only appeared when running my SvelteKit application in Docker, appears to have been a compatibility issue. I don't know the exact issue (beyond it being potentially TLS-related), but I was calling a Microsoft SQL Server 2014 server from Prisma. While this was working fine when running my application outside of a Docker container, it wasn't working in any environment once running in Docker.

My solution has been to migrate my database to a newer version of Microsoft SQL Server. [Here](https://www.prisma.io/docs/reference/database-reference/supported-databases) is the list of supported databases, for anyone who may be in a similar boat.

I tried getting around this by changing TLS options in Node...etc. Couldn't get anything to stick. I was due to migrate some databases to a new server anyway... 🤦‍♂️

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
